### ISSUE TITLE
Non-default ssh port

### ISSUE THREAD

**Issue by martinlindhe** [MichaelMure/git-bug#940]
I have a git repository on a ssh server that doesnt run on the default port 22.

It is configured with a Port entry in my ~/.ssh/config file, so the normal "ssh" command and the "git" command works:

```
Host srv
    Port 1234
```
The git remote just looks like this: "user@srv:repo"

Doing "git push" works, but trying to do a "git bug push", I end up with the following:

```
❯ git bug push
Error: dial tcp 192.168.1.5:22: connectex: No connection could be made because the target machine actively refused it.
```

I think it's an issue with a dependency.
(running in Windows & PowerShell)

**Comment by martinlindhe** [comment]
Nevermind, it seem to be unrelated issue.

**Comment by martinlindhe** [comment]
Actually, I am at loss. Can not seem to use any port except 22.

I just tested on macOS, and get a similar error:

```
Error: dial tcp 192.168.1.5:22: connect: connection refused
```

**Comment by MichaelMure** [comment]
Possibly related: https://github.com/go-git/go-git/issues/509

**Comment by martinlindhe** [comment]
> Possibly related: [go-git/go-git#509](https://github.com/go-git/go-git/issues/509)

You are right. I worked around the issue on my end for now by editing https://github.com/go-git/go-git/blob/f9b2cce5c9e6510fefb21ce54c07b59e83b6da8f/plumbing/transport/common.go#L244 and add a replace to go.mod

**Comment by DeeUnderscore** [comment]
Another workaround for this is to explicitly specify the port in your git remote, for example `ssh://git@example.com:1234/somerepo.git`. This seems to work with both git-bug and regular git.

**Comment by sudoforge** [comment]
@DeeUnderscore [said](https://github.com/git-bug/git-bug/issues/940#issuecomment-1336593875):
> Another workaround for this is to explicitly specify the port in your git remote, for example `ssh://git@example.com:1234/somerepo.git`. This seems to work with both git-bug and regular git.

As of this comment, this is the recommended approach for users who need to change the SSH port.

I'm going to slot this issue in for `0.8.2`.

**Comment by UlyssesApokin** [comment]
Is there a plan to fix this in the near future?

**Comment by sudoforge** [comment]
@UlyssesApokin we'd like to solve for this, but to be clear, doing so presents a large effort (as the git library we use does not support it), and "near" may be inaccurate.

that said, i think a viable solution for the near-term future is to shell out to `git` for push + pull operations. this has its own downsides, though, and is still something we are considering.

**Comment by UlyssesApokin** [comment]
Thanks, I solved this issue using the solution from https://github.com/git-bug/git-bug/issues/940#issuecomment-1336593875
I also had to transfer my repo from dumb http to smart http in case I want to share it and open it for reading.
It would probably be cool to switch to native git pull/push, considering that you need to run `git fetch "refs/identities/*:refs/identities/*"` to import user.
You're doing a great job. Thanks for the program. I hope you'll also add the ability to send and receive bugs via email, similar to `git format-patch`, `git send-email`, and `git am` or `git apply`.

**Comment by sudoforge** [comment]
> You're doing a great job. Thanks for the program. I hope you'll also add the ability to send and receive bugs via email, similar to git format-patch, git send-email, and git am or git apply.

thanks for the kind words! and yep, i saw [your comment](https://github.com/git-bug/git-bug/discussions/1391#discussioncomment-16484572) on "a vision for the future of git-bug". i'll follow up with thoughts there.
